pkg/ipamd/ipamd.go

```diff
@@ -2122,44 +2106,48 @@ func (c *IPAMContext) hasRoomForEni() bool {
 	return c.dataStore.GetENIs() < (c.maxENI - c.unmanagedENI - trunkEni)
 }
 
-func (c *IPAMContext) isDatastorePoolTooLow() bool {
-	short, _, warmTargetDefined := c.datastoreTargetState()
-	if warmTargetDefined {
-		return short > 0
+func (c *IPAMContext) isDatastorePoolTooLow() (bool, *datastore.DataStoreStats) {
+	stats := c.dataStore.GetIPStats(ipV4AddrFamily)
+	// If max pods has been reached, pool is not too low
```

Comment:
> If max pods has been reached, pool is not too low

I think, this comment is bit confusing, as max pods has been mentioned without the context. I got confused while reading this.

If we write,  if `c.maxPods <= stats.TotalIPs` then it seems alright.

Here we are saying, the nodes maxPods is less than or equal to the total ip available in the datastore, and so the isDatastorePoolTooLow too low is false.

Reply:
Yeah, what this function checks is a bit confusing. `isDatastorePoolTooLow` looks at the number of currently available IPs and compares that number to the warm/minimum targets. If there are not enough available IPs, we return false.

The `maxPods` check here is short-circuiting that logic by saying, "forgot about warm targets, do not allocate any more IPs if we have already allocated more than maxPods worth"

Reply:
I understood it better, when thinking like this 
> c.maxPods <= stats.TotalIPs

Because we are sending stats as the argument, I think, you went with the validation as  `stats.TotalIPs >= c.maxPods`; explaining the we have more than maxPods worth. 

Sounds good to me.